### PR TITLE
Add initial functionality

### PR DESCRIPTION
**Pull request by caternuson**
Basic I2C and SPI functionality in place.
**NOTE:** CAP1188 pins start at 1 - indexing matches pins.
```python
cap[1].value
cap[1].raw_value
```
Along with ability to recalibrate:
```python
cap.recalibrate()     # recal all pins
cap[1].recalibrate()  # recal just pin 1
```
**NOT INCLUDED**
* reset pin
* irq pin
* many other knobs

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+.. toctree::
+    :caption: API Reference
+    :maxdepth: 3
+
+    api
+
+.. toctree::
+    :caption: Tutorials
+
+.. todo:: Add any Learn guide links here. If there are none, then simply delete this todo and leave
```
Comment (sommersoft) [inline docs/index.rst]: Travis, once setup, will trip on this todo.

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    the toctree above for use later.
+
+.. toctree::
+    :caption: Related Products
+
+.. todo:: Add any product links here. If there are none, then simply delete this todo and leave
```
Comment (sommersoft) [inline docs/index.rst]: Same as above.

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    'sphinx.ext.intersphinx',
+    'sphinx.ext.napoleon',
+    'sphinx.ext.todo',
+]
+
+# TODO: Please Read!
+# Uncomment the below if you use native CircuitPython modules such as
+# digitalio, micropython and busio. List the modules you use. Without it, the
+# autodoc module docs will fail to generate with a warning.
+# autodoc_mock_imports = ["digitalio", "busio"]
```
Comment (sommersoft) [inline docs/conf.py]: You'll need to uncomment, and mock `micropython`.

File: docs/examples.rst [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Simple test
+------------
+
+Ensure your device works with this simple test.
+
+.. literalinclude:: ../examples/cap1188_simpletest.py
```
Comment (sommersoft) [inline docs/examples.rst:6]: This doesn't match the actual example files.

**Review by sommersoft — CHANGES_REQUESTED** [review]
Took a gander. Code looks good to me. Found just a few things to help Travis's mood once its setup.

For the example filenames, I _think_ the standard is to include "simpletest". But, that may be a call better left to @kattni.

Note: I **did not** test this, as I don't have the sensor available.

File: docs/examples.rst (hunk 4 above)
Comment (caternuson) [reply docs/examples.rst:6]: Example changed to be more standardy. Now it should.

**Comment by sommersoft** [comment]
@adafruit/circuitpythonlibrarians does anyone have a CAP1188 sensor to test this with? I sadly, do not...so my approval stops at that point. ☹️

File: adafruit_cap1188/cap1188.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+    @property
+    def manufacturer_id(self):
+        """The manufacturer ID."""
+        return self._read_register(CAP1188_REVISION)
+
+    @property
+    def revision(self):
+        """The revision number."""
+        return self._read_register(CAP1188_MANU_ID)
```
Comment (ladyada) [inline adafruit_cap1188/cap1188.py]: i think this is swapped with the manuf id above?

File: adafruit_cap1188/cap1188.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+    @property
+    def product_id(self):
+        """The product ID."""
+        return self._read_register(CAP1188_PRODUCT_ID)
+
+    @property
+    def manufacturer_id(self):
+        """The manufacturer ID."""
+        return self._read_register(CAP1188_REVISION)
```
Comment (ladyada) [inline adafruit_cap1188/cap1188.py]: swapped with rev register below?

**Review by ladyada — CHANGES_REQUESTED** [review]

**Comment by ladyada** [comment]
i can do the hardware test, but have some requested changes/fixes!

File: adafruit_cap1188/cap1188.py (hunk 5 above)
Comment (caternuson) [reply adafruit_cap1188/cap1188.py]: :man_facepalming:

File: adafruit_cap1188/cap1188.py (hunk 6 above)
Comment (caternuson) [reply adafruit_cap1188/cap1188.py]: :man_facepalming:

File: examples/cap1188_simpletest.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+from adafruit_cap1188.i2c import CAP1188_I2C
+i2c = busio.I2C(board.SCL, board.SDA)
+cap = CAP1188_I2C(i2c)
+
+# SPI setup
+# from digitalio import DigitalInOut, Direction
+# from adafruit_cap1188.spi import CAP1188_SPI
+# spi = busio.SPI(board.SCK, board.MOSI, board.MISO)
+# cs = DigitalInOut(board.D10)
+# cap = CAP1188_I2C(spi, cs)
```
Comment (ladyada) [inline examples/cap1188_simpletest.py]: you mean SPI here!

**Review by ladyada — CHANGES_REQUESTED** [review]

File: adafruit_cap1188/cap1188.py [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+    def recalibrate(self):
+        """Perform a self recalibration."""
+        self._cap1188.recalibrate_pins(1 << self._pin - 1)
+
+
+class CAP1188:
+    """CAP1188 driver base, must be extended for I2C/SPI interfacing."""
+    def __init__(self):
+        self._channels = [None]*8
```
Comment (ladyada) [inline adafruit_cap1188/cap1188.py:106]: check the manufacturer id and product id here to verify the chip is wired correctly

**Review by ladyada — CHANGES_REQUESTED** [review]

**Comment by ladyada** [comment]
id actually remove the properties for manufacturer, reveision etc and just check register values in init(), you'll never need that data again. you can see bme280 library for how we exception on failure

File: examples/cap1188_simpletest.py (hunk 7 above)
Comment (caternuson) [reply examples/cap1188_simpletest.py]: mmmm. copypasta.

File: adafruit_cap1188/cap1188.py (hunk 8 above)
Comment (ladyada) [reply adafruit_cap1188/cap1188.py:106]: thankx!

**Review by ladyada — APPROVED** [review]

**Comment by ladyada** [comment]
ok this library tests fine by me, good enough for v1.0.0 - we can add more stuff if people request it :)

**Review by kattni — APPROVED** [review]
Everything appears to be set with this, I'm going to go ahead and merge. Thanks @caternuson for doing this and thanks everyone for your reviews!

**Pull request merged**